File: app/src/main/java/com.example.safehome/ControlPanel/KeyPad.java
package com.example.safehome.ControlPanel;

public interface KeyPad {

	/**
	 * Defines the input from terminal, screen or key pad
	 * @param keyStrokes
	 */
	void readKeyStrokes(String keyStrokes);
	/**
	 * Decodes keystroke to to utf-8 encoding
	 * @return
	 */
	String decodeKey();




}

File: app/src/main/kotlin/com.example.safehome/ControlPanel/KeyPad.kt
package com.example.safehome.ControlPanel

/**
 * Represents user input
 */
interface KeyPad {
    /**
     * Defines the input from terminal, screen or key pad
     * @param keyStrokes the String keyStroke input.
     */
    fun readKeyStrokes(keyStrokes: String?)

    /**
     * Decodes keystroke to to utf-8 encoding
     * @return the decoded key String value
     */
    fun decodeKey(): String?
}